hwtracer/src/pt/mod.rs

```diff
@@ -1,3 +1,4 @@
+pub(crate) mod c_errors;
```

Comment:
Not all of the c_errors are PT-specific, but we can re-share those parts later if we ever implement another trace collector.

Reply:
Yes, I agree. In general, I think our error enums are weirdly unuseful, because they're so fine-grained that we push the complexity of the errors onto callers. Ironically, I think enums with fewer cases would be better for us. But I haven't fully thought that through, and it feels like something for a future PR to consider.

Reply:
I was trying to give better errors than "PT failed somehow" (one of the PT error codes) or "Something else in C failed" (an errno error code). Super generic errors waste time in debugging, tracking down where the error came from.

We might be able to do better, but it's not super important right now.

Reply:
I think, in general, `Box<dyn Error>` is the way to go, because you can put in an arbitrarily detailed string, but without giving the user the impression they can do much about the error. In the case of PT, we arguably only care about "an error that might go away if you try again" vs. "an error that's probably permanent".

Reply:
You surprise me! Once upon a time you used to dislike `dyn Error` due to the dynamic dispatch overhead.

Although we are talking about error paths, some are recoverable error paths, so performance may still matter.

I don't know how much `dyn` slows us down here though.

Reply:
> Once upon a time you used to dislike dyn Error due to the dynamic dispatch overhead.

I don't think I ever said that. Yes, dynamic dispatch has overhead, but for errors it's mostly a non-issue.